src/SOS/Strike/clrma/exception.cpp

```diff
@@ -0,0 +1,551 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+#include "managedanalysis.h"
+
+extern BOOL IsAsyncException(const DacpExceptionObjectData& excData);
+
+_Use_decl_annotations_
+ClrmaException::ClrmaException(ClrmaManagedAnalysis* managedAnalysis, ULONG64 address) :
+    m_lRefs(1),
+    m_managedAnalysis(managedAnalysis),
+    m_address(address),
+    m_typeName(nullptr),
+    m_message(nullptr),
+    m_exceptionDataInitialized(false),
+    m_stackFramesInitialized(false),
+    m_innerExceptionsInitialized(false)
+{
+    _ASSERTE(address != 0);
+    managedAnalysis->AddRef();
+}
+
+ClrmaException::~ClrmaException()
+{
+    TraceInformation("~ClrmaException\n");
+    if (m_typeName != nullptr)
```

Comment:
Minor nit - it would be nice to use smart pointers that would automatically clean up

Reply:
We don't really have a smart pointer for this case.